inductiva/simulators/fenicsx.py

```diff
@@ -15,25 +15,44 @@ def __init__(self):
     def run(
         self,
         input_dir: types.Path,
-        mesh_filename: str,
+        geometry_filename: str,
         bcs_filename: str,
         material_filename: str,
+        global_refinement_factor: float = 1.0,
+        local_refinement_factor: float = 0.0,
         machine_group: Optional[resources.MachineGroup] = None,
         run_async: bool = False,
     ) -> tasks.Task:
         """Run the simulation.
 
         Args:
-            mesh_filename: Mesh filename.
+            geometry_filename: Geometry filename.
             bcs_filename: Boundary conditions filename.
             material_filename: Material filename.
+            global_refinement_factor (float): The refinement factor for global
+              refinement of the mesh. A higher value results in a finer mesh
+              overall, increasing the number of elements in the entire mesh, and
+              leading to a more detailed representation of the geometry. Use
+              this factor when you want to globally refine the mesh uniformly,
+              without specific local focus.
+            local_refinement_factor (float): The refinement factor for local
+              refinement of the mesh. This factor controls the local refinement
+              level of the mesh and is typically used for refining specific
+              regions or features of the mesh. A higher value for this factor
+              indicates a finer mesh in the regions of interest, providing more
+              detailed resolution around certain features. Use this factor when
+              you want to focus on refining specific areas while keeping the
+              rest of the mesh less refined.
             machine_group: The machine group to use for the simulation.
             run_async: Whether to run the simulation asynchronously.
             other arguments: See the documentation of the base class.
         """
+
         return super().run(input_dir,
                            machine_group=machine_group,
                            run_async=run_async,
-                           mesh_filename=mesh_filename,
+                           geometry_filename=geometry_filename,
                            bcs_filename=bcs_filename,
-                           material_filename=material_filename)
+                           material_filename=material_filename,
+                           global_refinement_factor=global_refinement_factor,
```

Comment:
These parameters should just be passed to the Deformable Plate at this point... At least, for now they aren't used for the Linear.

You should be able to just pass them below with *kwargs or something.

Reply:
Not sure I understand or agree with this, @IvanPombo. These parameters control the resolution of the simulation, so they should be arguments here and not attributes of the plate class.

Reply:
On the Backend they are not used by the LinearElastic...Executer, so I am not sure they should be going here so explicitly. It assumes people using the low-level (if though it may not be perfect) can use this to modify the mesh.